packages/perseus/src/widgets/interactive-graphs/score-interactive-graph.ts

```diff
@@ -280,14 +281,11 @@ function scoreInteractiveGraph(
         }
     }
 
-    // The input wasn't correct, so check if it's a blank input or if it's
-    // actually just wrong
-    if (!hasValue || _.isEqual(userInput, rubric.graph)) {
-        // We're where we started.
-        return {
-            type: "invalid",
-            message: null,
-        };
+    // The input wasn't correct, so check if it's a blank input (validate) or if
+    // it's actually just wrong
+    const validationError = validateInteractiveGraph(userInput, scoringData);
+    if (validationError) {
+        return validationError;
```

Comment:
The validation function was put here since this is the only place where validation occurs. It seems like it should be okay to check if the user input is blank or if the user hasn't changed the graph yet at the start of scoring, but since that's not what was done and there is a comment specifically about doing this after confirming the input was not correct, I left it here.

Reply:
Realized that the none type graph check does need to happen before validation because all none type graphs will have empty input. Will need to review this before allowing validation to occur before scoring.